Library/PushNotificationDialog.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+public struct PushNotificationDialog {
+
+  public enum Context: String {
+    case login
+    case message
+    case pledge
+    case save
+  }
+
+  static public func canShowDialog(`for` context: PushNotificationDialog.Context) -> Bool {
+
+    return (!AppEnvironment.current.userDefaults.deniedNotificationContexts.contains(context.rawValue) &&
+            AppEnvironment.current.userDefaults.deniedNotificationContexts.count < 3) &&
```

Comment:
Why do you need to check the count is < 3?

Reply:
This is one of the rules to show the PushNotification dialog. We can  show the alerts only 3x (In case the user tapped `Not now` on the 2 first times).